homeassistant/config.py

```diff
@@ -263,12 +268,17 @@ def process_ha_config_upgrade(hass):
 
 
 @asyncio.coroutine
-def async_process_ha_core_config(hass, config):
+def async_process_ha_core_config(hass, config, complete_config=None):
     """Process the [homeassistant] section from the config.
 
     This method is a coroutine.
     """
     config = CORE_CONFIG_SCHEMA(config)
+
+    if complete_config:
+        merge_packages_config(complete_config, config[CONF_PACKAGES])
```

Comment:
`config.pop(CONF_PACKAGES)` and you won't have to delete it afterwards from the dict.